src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1520,4 +1520,44 @@ public static void SetTestHook(string property, object value)
             }
         }
     }
+
+    /// <summary>
+    /// An bounded stack based on a linked list.
+    /// </summary>
+    public class BoundedStack<T> : LinkedList<T>
```

Comment:
@PowerShell/powershell-committee needs to approve a new public api like this.

Reply:
Should we then rather think of making it internal (if possible)?

Reply:
I made it internal then.